charmcraft/commands/store/__init__.py

```diff
@@ -636,6 +636,31 @@ def _get_lib_info(*, full_name=None, lib_path=None):
         full_name=full_name, path=lib_path, lib_name=lib_name, charm_name=charm_name)
 
 
+def _get_libs_from_tree(charm_name=None):
+    """Get library info from the directories tree (for a specific charm if specified).
+
+    It only pays attention to the proper disk structures, if there.
+    """
+    local_libs_data = []
+
+    if charm_name is None:
+        base_dir = pathlib.Path('lib') / 'charms'
+        charm_dirs = sorted(base_dir.iterdir()) if base_dir.exists() else []
+    else:
+        base_dir = pathlib.Path('lib') / 'charms' / charm_name
+        charm_dirs = [base_dir] if base_dir.exists else []
+
+    for charm_dir in charm_dirs:
+        for v_dir in sorted(charm_dir.iterdir()):
+            if v_dir.is_dir() and v_dir.name[0] == 'v' and v_dir.name[1:].isdigit():
+                for libfile in sorted(v_dir.glob('*.py')):
```

Comment:
should this be recursive?

Reply:
Nop, a "fixed deep" is required as per the spec, not any tree, and each directory has different requirements, so...

Reply:
Thanks, I see that now with the libraries design :D